### PR TITLE
gh-614: Restored Viewer type-to-search feature.

### PR DESCRIPTION
**Pull request by MKadaner**
## Summary
On popular demand restored the Viewer type-to-search feature.

 
## References
#622

## Checklist
- [X] I have followed the [contributing guidelines](https://github.com/FarGroup/FarManager/blob/master/CONTRIBUTING.md).
- [X] I have discussed this with project maintainers: #614<br/>
If not checked, I accept that this work might be rejected in favor of a different grand plan.<br/>

## Details
As per the discussion, implemented as a macro.

### CODE REVIEW

File: far/dialog.hpp [hunk 1]
```diff
@@ -216,6 +216,7 @@ class Dialog: public window
 	void* GetDialogData() const {return DataDialog;}
 	void InitDialog();
 	void Process();
+	void ProcessKey(int Key, size_t ItemPos);
```
Comment (MKadaner) [inline far/dialog.hpp:219]: Not required for the variant with `std::optional<Manager::Key>`. Need to revert if we go with it.

File: far/stddlg.cpp [hunk 2]
```diff
@@ -258,6 +259,16 @@ int GetSearchReplaceString(
 	{
 		switch (Msg)
 		{
+		case DN_DRAWDIALOGDONE:
+			{
+				if (FirstKey)
+				{
+					Dlg->ProcessKey(*FirstKey);
```
Comment (MKadaner) [inline far/stddlg.cpp:266]: This used to be
https://github.com/FarGroup/FarManager/blob/833c93989cc63b3fde1a9f85552bd4d730db2b7b/far/viewer.cpp#L2594

Does it make any sense? Do I need these complications?

**Comment by alabuzhev** [comment]
> POC only, for gathering feedback on the two possible implementations.

Let's wait and gather user feedback first.
Perhaps the third possible implementation (a simple macro) will do.

**Comment by MKadaner** [comment]
Makes perfect sense to me. Please ping me if you decide to complete this PR or just close it. Not that I do not have my own opinion, but I still do not hold it much above that of another user.

**Comment by MKadaner** [comment]
Done a macro. Please review.

**Review by alabuzhev — APPROVED** [review]

**Comment by MKadaner** [comment]
Rebased, ready. Thanks.

**Comment by alabuzhev** [comment]
Thank you.

**Pull request merged**